source/types/Case.ts

```diff
@@ -0,0 +1,267 @@
+  /**
+   * Unique ID for this field in the form.
+   */
+  id: string;
+
+  /**
+   * List of tags with generic use.
+   *
+   * For VIVA this is used to connect and transform answers to VIVA properties in VADA.
+   */
+  tags?: string[];
+}
+
+export interface Answer {
+  /**
+   * Details of the field for this answer.
+   */
+  field: AnswerField;
+
+  /**
+   * The actual value of this answer.
+   */
+  value: string | boolean;
+}
+
+export interface EncryptedAnswersWrapper {
+  /**
+   * Encrypted string output of the answers object.
+   */
+  encryptedAnswers: string;
+}
+
+export interface FormPosition {
+  currentMainStep: number;
+  currentMainStepIndex: number;
+  index: number;
+  level: number;
+}
+
+export interface AnsweredForm {
+  /**
+   * Answers for this form.
+   *
+   * If encrypted then this is an object, otherwise this is an array.
+   */
+  answers: Answer[] | EncryptedAnswersWrapper;
+
+  /**
+   * Last saved position of this form. Used to return to the same page when resuming a form.
+   */
+  currentPosition: FormPosition;
+
+  /**
+   * Details of the encryption used for this form.
+   *
+   * If unencrypted, `encryption.type` is set to `decrypted`.
+   */
+  encryption: EncryptionDetails;
+}
+
+export interface Person {
+  /**
+   * First name of this person.
+   */
+  firstName: string;
+
+  /**
+   * `true` if this person has signed the case, or `false` if they have yet to sign.
+   */
+  hasSigned?: boolean;
+
+  /**
+   * Last name of this person.
+   */
+  lastName: string;
+
+  /**
+   * Swedish personal identity number (personnummer) of this person.
+   */
+  personalNumber: string;
+
+  /**
+   * Role of this person in relation to the case.
+   */
+  role: PersonRole;
+}
+
+export interface Status {
+  /**
+   * User-friendly description of this status.
+   */
+  description: string;
+
+  /**
+   * User-friendly name of this status.
+   */
+  name: string;
+
+  /**
+   * Unique identifier of this status.
+   */
+  type: ApplicationStatusType;
+}
+
+export interface Case {
+  /**
+   * Unix timestamp in milliseconds for when this case object was first created.
+   */
+  createdAt: number;
+
+  /**
+   * Form UUID for the currently active form in `forms`.
+   */
+  currentFormId: string;
+
+  /**
+   * 3rd party provider details for this case.
+   */
+  details: VIVACaseDetails;
+
+  /**
+   * Unix timestamp in seconds for when this case expires and will be removed from the database.
+   */
+  expirationTime: number;
+
+  /**
+   * Map of forms connected to this case.
+   */
+  forms: {
+    [formId: string]: AnsweredForm;
+  };
+
+  /**
+   * UUID of this case.
+   */
+  id: string;
+
+  /**
+   * Binary PDF data of the PDF generated from this case.
+   */
+  pdf?: Buffer;
+
+  /**
+   * `true` if pdf has been generated.
+   */
+  pdfGenerated?: boolean;
+
+  /**
+   * List of persons connected to this case.
+   */
+  persons: Person[];
+
+  /**
+   * Primary key; usually "USER#[applicant pno]"
+   */
```

Comment:
Primary key is a combination of PK and SK to be correct according to DynamoDB's documentation
In this case PK is rather called after DynamoDB lingual Partition Key